exec/java-exec/src/main/java/org/apache/drill/exec/store/log/LogRecordReader.java

```diff
@@ -0,0 +1,623 @@
+      Pattern r = Pattern.compile("^field_(\\d+)$");
+
+      // Now create matcher object.
+      Matcher m = r.matcher(name);
+      int patternIndex = -3;
+
+      if (m.find()) {
+        //if no fields are defined in the configuration, then all the fields have names of 'field_n'
+        //Therfore n is the column index
+        patternIndex = Integer.parseInt(m.group(1));
+      } else {
+        for (int i = 0; i < fields.size(); i++) {
+          if (fields.get(i).equalsIgnoreCase(name)) {
+            patternIndex = i;
+            break;
+          } else if( name.equals("_raw")){
+            patternIndex = -1;
+            break;
+          } else if( name.equals("_unmatched_rows")){
+            patternIndex = -2;
+            unmatchedColumnIndex = colIndex;
+          }
+        }
+      }
+      columns[colIndex++] = makeColumn(name, patternIndex);
+    }
+  }
+
+  private ColumnDefn makeColumn(String name, int patternIndex) {
+    String typeName = null;
+    if (patternIndex <= -1) {
+      // Use VARCHAR for missing columns
+      // (instead of Drill standard of nullable int)
+      typeName = MinorType.VARCHAR.name();
+    } else if (patternIndex < formatConfig.dataTypes.size()) {
+      typeName = formatConfig.dataTypes.get(patternIndex);
+    }
+    if (typeName == null) {
+      // No type name. VARCHAR is a safe guess
+      typeName = MinorType.VARCHAR.name();
+    }
+    if( name.equals("_raw")){
+      return new VarCharDefn(name, patternIndex);
+    }
+
+    MinorType type = MinorType.valueOf(typeName);
+    switch (type) {
+      case VARCHAR:
+        return new VarCharDefn(name, patternIndex);
+      case INT:
+      case SMALLINT:
+      case BIGINT:
+        return new IntDefn(name, patternIndex);
+      case FLOAT8:
+      case FLOAT4:
+      case DECIMAL9:
+      case DECIMAL18:
+        return new DoubleDefn(name, patternIndex);
+      case DATE:
+        return new DateDefn(name, patternIndex, formatConfig.getDateFormat());
+      case TIMESTAMP:
+        return new TimeStampDefn(name, patternIndex, formatConfig.getDateFormat());
+      case TIME:
+        return new TimeDefn(name, patternIndex, formatConfig.getTimeFormat());
+      default:
+        throw UserException
+            .validationError()
+            .message("Undefined column types")
+            .addContext("Position", patternIndex)
+            .addContext("Field name", name)
+            .addContext("Type", typeName)
+            .build(logger);
+    }
+  }
+
+  private void defineVectors() {
+    for (int i = 0; i < columns.length; i++) {
+      try {
+        columns[i].define(outputMutator);
+      } catch (SchemaChangeException e) {
+        throw UserException
+            .systemError(e)
+            .message("Vector creation failed")
+            .build(logger);
+      }
+    }
+  }
+
+  @Override
+  public int next() {
+    rowIndex = 0;
+    while (nextLine()) {
+    }
+    return rowIndex;
+  }
+
+  private boolean nextLine() {
```

Comment:
The code above mentions a `_raw` column, but I don't see it being filled in anywhere here.

Should the `_raw` column be set only for matched rows? For both matched and unmatched rows? See discussion above.